### PR TITLE
Add transform check + tests

### PR DESCRIPTION
**Pull request by AleDore**
This PR enables check on `<ds:Transform>` occurrences number, in order to avoid parsing unexpected SAML Response and Assertion.

### CODE REVIEW

File: src/utils/__tests__/saml.test.ts [hunk 1]
```diff
@@ -193,8 +197,77 @@ describe("preValidateResponse", () => {
     });
   });
 
+  it("should preValidate fail when saml Assertion has more than 4 Transform elements in SignedInfo", async () => {
```
Comment (BurnedMarshal) [inline src/utils/__tests__/saml.test.ts:200]:
```suggestion
  it("should preValidate fail when saml response has more than 4 Transform (4 Assertion + 2 Response) elements in SignedInfo", async () => {
```

File: src/utils/__tests__/saml.test.ts [hunk 2]
```diff
@@ -193,8 +197,77 @@ describe("preValidateResponse", () => {
...
+    await asyncExpectOnCallback(mockCallback, expectedError);
+    expect(mockEventTracker).toBeCalledWith({
+      data: {
+        message: expectedError.message
+      },
+      name: expectedGenericEventName,
+      type: "ERROR"
+    });
+  });
+  it("should preValidate fail when saml Response has more than 4 Transforms elements in SignedInfo", async () => {
```
Comment (BurnedMarshal) [inline src/utils/__tests__/saml.test.ts:237]:
```suggestion
  it("should preValidate fail when saml response has more than 4 Transform elements (4 Response + 2 Assertion) in SignedInfo", async () => {
```

File: src/utils/saml.ts [hunk 3]
```diff
@@ -583,6 +584,32 @@ const isEmptyNode = (element: Element): boolean => {
...
+  )(
+    Array.from(
+      targetElement.getElementsByTagNameNS(SAML_NAMESPACE.XMLDSIG, "Transform")
+    )
+  ).foldL(
+    () => right(targetElement),
+    transformElements =>
+      fromPredicate(
+        (_: readonly Element[]) => !isOverflowNumberOf(_, 4),
+        () => new Error("Transform cannot occurs more than 4 occurrences")
```
Comment (BurnedMarshal) [inline src/utils/saml.ts:608]:
```suggestion
        () => new Error("Transform element cannot occurs more than 4 times")
```

**Review by BurnedMarshal — DISMISSED** [review]
LGTM apart minor changes

**Review by BurnedMarshal — DISMISSED** [review]

**Review by pp-ps — DISMISSED** [review]

File: src/utils/saml.ts [hunk 4]
```diff
@@ -1369,18 +1385,38 @@ export const getPreValidateResponse = (
...
       error => {
         if (eventHandler) {
-          eventHandler({
-            data: {
-              message: error.message
-            },
-            name: "spid.error.generic",
-            type: "ERROR"
-          });
+          error instanceof Error
```
Comment (BurnedMarshal) [inline src/utils/saml.ts:1393]: can we use `TransformError.is` instead of `instanceof`?

**Review by BurnedMarshal — APPROVED** [review]
LGTM apart a minor change

**Review by BurnedMarshal — APPROVED** [review]

**Pull request merged**